test/wsfed.tests.js

```diff
@@ -94,6 +99,34 @@ describe('wsfed', function () {
     });
   });
 
+  describe('when using a different name identifier format', function (){
+    var body, $, signedAssertion, attributes;
+
+    const fakeNameIdentifierFomat = 'urn:oasis:names:tc:SAML:1.1:nameid-format:swfedfakeformat';
+
+    before(function (done) {
+      server.options = { nameIdentifierFormat: fakeNameIdentifierFomat };
+      request.get({
+        jar: request.jar(),
+        uri: 'http://localhost:5050/wsfed?wa=wsignin1.0&wctx=123&wtrealm=urn:the-super-client-id'
+      }, function (err, response, b){
+          if(err) return done(err);
+          body = b;
+          $ = cheerio.load(body);
+          var wresult = $('input[name="wresult"]').attr('value');
+          signedAssertion = /<t:RequestedSecurityToken>(.*)<\/t:RequestedSecurityToken>/.exec(wresult)[1];
+          attributes = xmlhelper.getAttributes(signedAssertion);
+          done();
+      });
+    });
+
+    it(`should set name identifier format to ${fakeNameIdentifierFomat}`, function (){
```

Comment:
And maybe this scenario can be also a bit more explicit, somelithing like `should set name identifier format passed as an auth option`, otherwise it might not be clear from checking test results why the name identifier format is supposed to be set to `urn:oasis:names:tc:SAML:1.1:nameid-format:swfedfakeformat`